keps/1618-optional-gc-of-workloads/README.md

```diff
@@ -0,0 +1,252 @@
+are designed to be explicitly enabled and configured by users, ensuring backward compatibility. 
+While the primary focus is on Workloads, the API section can be extended to include 
+other Kueue-authored objects in the future. The implementation involves incorporating the 
+deletion logic into the reconciliation loop and includes considerations for 
+potential risks, such as handling a large number of existing Workloads. 
+Overall, this KEP aims to enhance Kueue's resource management and provide users with 
+more control over object retention.
+
+## Motivation
+
+Currently, Kueue does not delete its own Kubernetes objects, leaving it to Kubernetes' 
+garbage collector (GC) to manage the retention of objects created by Kueue, such as Pods, Jobs, and Workloads. 
+We do not set any expiration on these objects, so by default, they are kept indefinitely. Based on usage patterns, 
+including the amount of jobs created, their complexity, and frequency over time, these objects can accumulate, 
+leading to unnecessary use of etcd storage and a gradual increase in Kueue's memory footprint. Some of these objects, 
+like finished Workloads, do not contain any useful information that could be used for additional purposes, such as debugging. 
+That's why, based on user feedback in [#1618](https://github.com/kubernetes-sigs/kueue/issues/1618), a mechanism for garbage collection of finished Workloads 
+is being proposed.
+
+With this mechanism in place, deleting finished and deactivated Workloads has several benefits:
+- **Deletion of superfluous information**: Finished Workloads do not store any useful runtime-related information 
+(which is stored by Jobs), nor are they useful for debugging. They could possibly be useful for auditing purposes, 
+but this is not a primary concern.
+- **Decreasing Kueue memory footprint**: Finished and deactivated Workloads are loaded during Kueue's initialization and kept in memory, 
+consuming resources unnecessarily.
+- **Freeing up etcd storage**: Finished and deactivated Workloads are stored in etcd memory until Kubernetes' built-in GC mechanism 
+collects them, but by default, Custom Resources are stored indefinitely.
+
+### Goals
+
+- Support the deletion of finished and deactivated Workload objects.
+- Introduce configuration of global retention policies for Kueue-authored Kubernetes objects, 
+- Maintain backward compatibility (the feature must be explicitly enabled and configured).
+
+### Non-Goals
+
+- Support the deletion/expiration of jobs or other Kubernetes objects not authored by Kueue.
+
+## Proposal
+
+Add a new field called `objectRetentionPolicies` to the Kueue Configuration API, 
+which will enable specifying a retention policy for finished and deactivated Workloads first under a
+`WorkloadRetentionPolicy` section. This API would be extended in the future 
+with options for other Kueue-managed objects.
+
+
+### User Stories
+
+#### Story 1 - Configurable retention for finished and deactivated Workloads
```

Comment:
I would this as separate user stories, since they originated from two different issues.

Reply:
Have split this in two user stories